pkg/inframetadata/reporter.go

```diff
@@ -137,6 +147,36 @@ func (r *Reporter) ConsumeResource(res pcommon.Resource) error {
 	return nil
 }
 
+func (r *Reporter) ConsumeMetrics(md pmetric.Metrics) error {
```

Comment:
Does this func need to be exported? If so considering adding docs

Reply:
Yes, documented it! The idea is to call this on the exporter when consuming metrics